fakeredis/commands_mixins/json_mixin.py

```diff
@@ -0,0 +1,605 @@
+path_pattern: re.Pattern = re.compile(r"^((?<!\$)\.|(\$\.$))")
+
+
+def format_jsonpath(path: Union[str, bytes]) -> str:
+    """Format the supplied JSON path value."""
+    if isinstance(path, bytes):
+        path = path.decode()
+
+    return path_pattern.sub("$", path)
+
+
+def null_op(obj: JsonType, *_: Any, **__: Any) -> None:
+    """Return the supplied object, completely unaltered."""
+    return obj
+
+
+class JSONObject:
+    """Argument converter for JSON objects."""
+
+    DECODE_ERROR = msgs.WRONGTYPE_MSG
+    ENCODE_ERROR = msgs.WRONGTYPE_MSG
+
+    @classmethod
+    def valid(cls, value: Any) -> bool:
+        """Determine if the supplied value is a "valid" `JSONObject`."""
+        raise NotImplementedError
+
+    @classmethod
+    def decode(cls, value: bytes) -> Any:
+        """Deserialize the supplied bytes into a valid Python object."""
+        return json.loads(value)
+
+    @classmethod
+    def encode(cls, value: Any) -> bytes:
+        """Serialize the supplied Python object into a valid, JSON-formatted
+        byte-encoded string."""
+        raise NotImplementedError
+
+
+class JSONCommandsMixin:
+    """`CommandsMixin` for enabling RedisJSON compatibility in `fakeredis`."""
+
+    def __init__(
+        self: FakeSocket,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        pass
+
+    @command(
+        name="JSON.DEL",
+        fixed=(Key(),),
+        repeat=(bytes,),
+    )
+    def json_del(
+        self: FakeSocket,
+        key: CommandItem,
+        path: Optional[bytes] = None,
+    ) -> int:
+        """Delete the JSON value stored at key `key` under `path`.
+
+        For more information see `JSON.DEL
+        <https://redis.io/commands/json.del>`_.
+        """
+        cached_value = json.loads(key.value or b"null")
+
+        if cached_value is None:
+            return 0
+
+        path = format_jsonpath(path or b"$")
+
+        if path != "$":
+            raise NotImplementedError("Path-based key-value deletion not yet supported!")
+
+        key.value = None
+        key.writeback()
+
+        return 1
+
+    # `forget` is an alias for `delete`
+    json_forget = json_del
+
+    @command(
+        name="JSON.GET",
+        fixed=(Key(),),
+        repeat=(bytes,),
+    )
+    def json_get(
+        self: FakeSocket,
+        name: CommandItem,
+        *args: bytes,
+    ) -> bytes:
+        """Get the object stored as a JSON value at key `name`.
+
+        `args` is zero or more paths, and defaults to root path
+        ``no_escape` is a boolean flag to add no_escape option to get
+        non-ascii characters
+
+        For more information see `JSON.GET <https://redis.io/commands/json.get>`_.
+        """
+        no_escape, args = (
+            bool(b"noescape" in args),
+            tuple(
+                map(
+                    parse_jsonpath,
+                    (format_jsonpath(arg) for arg in args if arg != b"noescape"),
+                )
+            ),
+        )
```

Comment:
It is a bit complex and does not support `noescape` in uppercase, etc., why not:
```python
no_escape = any([casematch(b'noescape', arg) for arg in args]) # Though not really used based on the next lines..
paths = tuple(
      parse_jsonpath(format_jsonpath(arg)) 
      for arg in args 
      if not casematch(b'noescape', arg))
```